### PR TITLE
Sphinx documentation

### PR DESCRIPTION
**Pull request by arh89**
#13 Implement Sphinx to automatically generate documentation

* Documentation can be built locally (provided Sphinx is installed) with `cd doc && make html`
* Documentation is built (as a test) using Travis
* Any commits to master will automatically rebuild documentation on readthedocs: http://improver.readthedocs.io/

### CODE REVIEW

**Comment by arh89** [comment]
I just noticed that the RTD hosted documentation has an "Edit on Github" link which results in a 404 error on most pages - this is because they link to the *.rst files in the repo, which are not present because they are instead generated at build time.

Note: Local builds use the Sphinx default theme (rather than the RTD theme), and so this won't be a problem.

Apparently, we can fix this problem by copying one of the template files from the RTD theme to our repo:
https://stackoverflow.com/questions/36019670/removing-the-edit-on-github-link-when-using-read-the-docs-sphinx-with-readthed

...I think that this fix is quite messy though. Including a single file to fix something which is not actually a part of the repo doesn't seem like an elegant solution.

File: bin/improver-tests [hunk 1]
```diff
@@ -45,6 +45,12 @@ echo_ok "pep8"
 ${PYLINT:-pylint} -E --rcfile=../etc/pylintrc improver
 echo_ok "pylint -E"
 
+# Build documentation as test.
+cd $IMPROVER_DIR/doc
+make html
```
Comment (benfitzpatrick) [inline bin/improver-tests]: Can you quiet this so that it doesn't produce output (unless it fails!)?

File: ACKNOWLEDGEMENTS.md [hunk 2]
```diff
@@ -3,6 +3,11 @@ Credit for external pieces of work:
...
   - .travis.yml, derived from Iris travis setup
   - .gitignore used as basis for ours
+  - environment.yml derived from Iris minimal conda environment
 
 BATS (https://github.com/sstephenson/bats), MIT-style:
   - tests/bin/bats\*, unaltered, from commit 0360811
+
+Automated generation of API documentation from sphinx-build:
+  - Based on solution posted by BowenFu:
+    https://github.com/rtfd/readthedocs.org/issues/1139
```
Comment (benfitzpatrick) [inline ACKNOWLEDGEMENTS.md]: Probably also add in the Makefile and conf.py origin

**Comment by benfitzpatrick** [comment]
Two comments, otherwise fine apart from the Travis failure - try a rebase?

File: ACKNOWLEDGEMENTS.md (hunk 2 above)
Comment (arh89) [inline ACKNOWLEDGEMENTS.md]: I thought I'd done that - turns out I had, but I'd not added it to the commit - doh!

**Comment by arh89** [comment]
@benfitzpatrick, merged in the changes from master and still couldn't figure out why Travis was failing - it turned out that I'd not updated all of the BATS tests for improver-tests (I added a reference to documentation in the script usage string).

...Hopefully that should fix things.

**Comment by benfitzpatrick** [comment]
Raised #89 for addressing these Sphinx warnings by altering the docstrings

**Comment by benfitzpatrick** [comment]
Could we make sure it is either `IMPROVER` (project) or `improver` (codebase), and include the following blurb somewhere noticeable:
`IMPROVER is a library of algorithms for meteorological post-processing and verification.`

Otherwise good.

**Comment by arh89** [comment]
@benfitzpatrick How's that?

**Comment by benfitzpatrick** [comment]
I think the 'edit on GitHub' bit is probably unnecessary to fix right now, let's leave it.

**Comment by arh89** [comment]
@gavinevans: Following a conversation with @benfitzpatrick, the Makefile has been updated to make it clear that it was automatically generated by Sphinx (as the other files already do). This makes it absolutely clear that we are not attempting to relicense any part of/any files generated by Sphinx.

**Pull request merged**